libraries/Network/src/NetworkUdp.cpp

```diff
@@ -292,7 +292,10 @@ void NetworkUDP::flush() {}
 
 int NetworkUDP::parsePacket() {
   if (rx_buffer) {
-    return 0;
+    if (rx_buffer->full()) { // packet was not read yet
+      return rx_buffer->available();
+    }
+    clear(); // discard the rest of the packet
```

Comment:
What will happen in you call `parsePacket` from one thread, but you read it's data from another?